_posts/2016-08-11-React-Native-at-Artsy.markdown

```diff
@@ -0,0 +1,173 @@
+
+* There are currently many more people trying to get help, with often arguably simple questions in the context of iOS
+  development, compared to those willing to spend time on answering questions. Most fora I’ve seen suffer from the
+  tragedy of the commons problem, which can be a real problem if you have incidental framework specific questions, but
+  React-Native probably is a complete disaster if you know very little about iOS development yet.
+
+  If you’re not already knowledgable about iOS development, are not motivated to put in the time to learn about the
+  platform specific details, and think making rich iOS applications with React Native will be a breeze, you’ll
+  [come home from a very cold fair](http://www.dwotd.nl/2008/06/443-van-een-kouwe-kermis-thuiskomen.html) indeed. While
+  you can definitely make applications that way, in my experience those often end up not feeling like proper citizens of
+  the platform they inhibit.
+
+# Pros, why we wanted to use React Native
+
+* From [the React website](https://facebook.github.io/react/):
+  > React will efficiently update and render just the right components when your data changes.
+  > Build encapsulated components that manage their own state, then compose them to make complex UIs.
+
+  The functional model that React introduces that allows you to reason about the state of your views in much simpler
+  ways has for us been a welcome change that should make it much easier to write decoupled code going forward.
+
+  I’m by no means a functional programming purist, nor do I really care for being one. I found the React/React Native
+  communities to be very welcome to functional enthusiasts and pragmatists alike, a healthy mix that I find leads to
+  more productive outcomes.
+
+* Relay. From [its website](https://facebook.github.io/relay/):
+  > Queries live next to the views that rely on them, so you can easily reason about your app.
+  > Relay aggregates queries into efficient network requests to fetch only what you need.
+
+  The clarity this brings to the view codebase - coupled with its smart caching, networking is just ground-breaking. 
+  No more multiple levels of model code in your application that you need to trace, just a single file with
+  [the view component](https://github.com/artsy/emission/blob/a2e4dbdb/lib/components/artist/header.js#L87) _and_
+  [the data it needs](https://github.com/artsy/emission/blob/a2e4dbdb/lib/components/artist/header.js#L143-L144). Neat.
+
+  We do still have ‘view models’, however, those now pretty much all live in
+  [our GraphQL service](http://artsy.github.io/blog/2016/06/19/graphql-for-mobile/). The added benefit here is that we
+  share that model logic with Artsy’s other (web) clients.
+
+* “Learn once, write anywhere.” is the neo-cross-platform slogan deployed by React, which is a play on the tried
+  “Write once, run anywhere.” slogan of yesteryear. The difference being that you can use the same paradigms to create
```

Comment:
I didn't even know that yesteryear was an actual word!

![screen shot 2016-08-10 at 5 33 01 pm](https://cloud.githubusercontent.com/assets/542335/17571980/8164e8ba-5f20-11e6-9dc3-f786bf25a441.png)